sktime/utils/git_diff.py

```diff
@@ -0,0 +1,80 @@
+"""Git related utilities to identify changed modules."""
+
+__author__ = ["fkiraly"]
+__all__ = []
+
+import importlib.util
+import inspect
+import subprocess
+
+
+def get_module_from_class(cls):
+    """Get full parent module string from class.
+
+    Parameters
+    ----------
+    cls : class
+        class to get module string from, e.g., NaiveForecaster
+
+    Returns
+    -------
+    str : module string, e.g., sktime.forecasting.naive
+    """
+    module = inspect.getmodule(cls)
+    return module.__name__ if module else None
+
+
+def get_path_from_module(module_str):
+    r"""Get local path string from module string.
+
+    Parameters
+    ----------
+    module_str : str
+        module string, e.g., sktime.forecasting.naive
+
+    Returns
+    -------
+    str : local path string, e.g., sktime\forecasting\naive.py
+    """
+    try:
+        module_spec = importlib.util.find_spec(module_str)
+        if module_spec is None:
+            raise ImportError(
+                f"Errpr in get_path_from_module, module '{module_str}' not found."
+            )
+        return module_spec.origin
+    except Exception as e:
+        raise ImportError(f"Error finding module '{module_str}': {e}")
+
+
+def is_module_changed(module_str):
+    """Check if a module has changed compared to the main branch.
+
+    Parameters
+    ----------
+    module_str : str
+        module string, e.g., sktime.forecasting.naive
+    """
+    module_file_path = get_path_from_module(module_str)
+    cmd = f"git diff remotes/origin/main -- {module_file_path}"
```

Comment:
If you're checking against remote branch, do we need to pull `main`? Can shallow repositories not compare against remote branches?

Reply:
May be not, got confused after seeing this: https://stackoverflow.com/questions/23708231/git-shallow-clone-clone-depth-misses-remote-branches

Reply:
I tried many things, and this is the only one that worked...

I am by no means a `git` expert, so if you have a better idea, feel free to try (preferably in a separate branch unless you are 100% sure it works)

Reply:
I think this is also resolved by adopting @yarnabrina's alternative suggestion by merge to this PR?

Reply:
Yes, it's resolved as well.